[FacebookBridge] Facebook bridge fix, swap to touch.facebook for groups

SpangleLabs opened the pull request:
This fixes group feeds, I think. Tested on a few groups.
It's swapping over to parsing using touch.facebook.com for groups. Users are still using www.facebook.com
I've got an automated integration test in here too, not sure how you'll feel about that. I'm curious how Travis likes it, as I've only ran it manually in PHP7.

Sorry this took a little while, I've had a very busy week moving house.

(If you're happy this PR is not spam, I would appreciate a `hacktoberfest-accepted` tag, but not required)

Commit:
[FacebookBridge] Fixing comment typo

Commit:
[FacebookBridge] Allow touch.facebook.com URLs also

Commit:
[FacebookBridge] Get HTML from the touch.facebook.com page

Commit:
[FacebookBridge] Fixing up isPublicGroup() method for new touch.facebook.com based parsing

Commit:
[FacebookBridge] Fixing up extractGroupName() method for new touch.facebook.com based parsing

Commit:
[FacebookBridge] Renaming group post parsing methods for clarity

Commit:
[FacebookBridge] Getting the right div for posts, after switch to touch.facebook.com

Commit:
[FacebookBridge] Fixing up post parsing after switch to using touch.facebook.com for groups

SpangleLabs commented:
This should close #1810 and #1691 

(On a side note, #1745 was transient, and looks like it can be closed now)

em92 commented on bridges/FacebookBridge.php:
```diff
@@ -186,19 +192,19 @@ private function collectGroupData() {
 
 			$item = array();
 
-			$item['uri'] = $this->extractGroupURI($post);
-			$item['title'] = $this->extractGroupTitle($post);
-			$item['author'] = $this->extractGroupAuthor($post);
-			$item['content'] = $this->extractGroupContent($post);
-			$item['timestamp'] = $this->extractGroupTimestamp($post);
-			$item['enclosures'] = $this->extractGroupEnclosures($post);
+			$item['uri'] = $this->extractGroupPostURI($post);
+			$item['title'] = $this->extractGroupPostTitle($post);
+			$item['author'] = $this->extractGroupPostAuthor($post);
+			$item['content'] = $this->extractGroupPostContent($post);
+			$item['timestamp'] = $this->extractGroupPostTimestamp($post);
+			$item['enclosures'] = $this->extractGroupPostEnclosures($post);
```
Why do we need to rename methods?

SpangleLabs replied:
I felt it made things more clear, and given they're internal methods, it shouldn't make any difference elsewhere. 
`extractGroupName()` and `extractGroupTitle()` have very similar and confusing method names otherwise. It took me a moment to realise the former is the group title, and the latter is the group-post title

em92 replied:
Fair enough

em92 commented on lib/contents.php:
```diff
@@ -233,7 +233,7 @@ function getContents($url, $header = array(), $opts = array(), $returnHeader = f
  * @param string $defaultSpanText Specifies the replacement text for `<span />`
  * tags when returning plaintext.
- * @return string Contents as simplehtmldom object.
+ * @return false|simple_html_dom Contents as simplehtmldom object.
```
Not related to Facebook bridge

SpangleLabs replied:
Shall I file another PR for that little type hint fix then? I'm not sure why it was labelled as returning a string tbh, but it makes my IDE sad

em92 commented on tests/bridges/FacebookBridgeTest.php:
```diff
@@ -0,0 +1,102 @@
+\Configuration::verifyInstallation();
+\Configuration::loadConfiguration();
+
+class FacebookBridgeTest extends TestCase
```
Nice idea about Facebook tests. I suggest to place them to other PR, 'because it is required to discuss it.

SpangleLabs replied:
That makes sense, I'll pull them out elsewhere.
I found them very helpful for development, to make sure things didn't break elsewhere and that they worked in multiple groups. Seems like Travis didn't fail due to them which is good, but I'm not certain whether it actually ran them

SpangleLabs commented:
There we go, moved the integration tests to another branch with PR #1823 and pulled out the simple html dom type hint fix, added a couple more type hint fixes, and filed as PR #1824

em92 commented:
gj @joshcoales
